### ISSUE TITLE
Update `linkerd doc` output to include the opaque-ports annotation

### ISSUE THREAD

**Issue by adleong** [linkerd/linkerd2#5894]
The `linkerd doc` command outputs a list called `AnnotationsReference` which describes all of the pod annotations which can be used to configure the proxy-injector.  This list appears to be missing the `config.linkerd.io/opaque-ports` annotation.  We should update this command to include that annotation.

Furthermore, we should ensure there are no other missing annotations and that this documentation is otherwise up to date.

**Comment by adleong** [comment]
To clarify that this data was automatically generated, we should also add a comment to the output to this effect.  E.g.

```
# Automatically generated by the linkerd doc command, do not manually edit
```

**Comment by alpeb** [comment]
Fixed by #5906
